### DISCUSSION TITLE
Add carat range to peerDep in query devtools

### DISCUSSION THREAD

**Discussion by dbasilio** [TanStack/query#5832] [General]
Hello! The `peerDependency` from `@tanstack/react-query-devtools` is pinned to a version of react-query. We ship a small library for our teams to use that includes the devtools as a dependency, but lists react-query as a `peerDependency` so that teams can upgrade the version without needing our involvement. 

This leads to failing peerDependencies though. If a team tries to update `@tanstack/react-query`, it fails the pinned version from `@tanstack/react-query-devtools`. Since that library is brought in transitively, it's not even listed in their dependencies, so you have to update the resolutions which is an extra step that some teams are unfamiliar with doing (we use Yarn v3 if that helps your understanding of how to manually update resolutions.

We can try to solve this on our end in a few ways (listing the devtools as a peerDep so it's directly installed), but I was wondering if you would be open to changing the range to a carat one which would get around the problem entirely?

**Comment by TkDodo** [comment]
yeah I think we can do that. we use [pnpm workspaces](https://pnpm.io/workspaces#publishing-workspace-packages), so I think the dependency must go from `"workspace:*"` to `"workspace:^"`, at least according to the docs.

would you like to PR it? I think this would apply to all packages that have peerDependencies depending on another package in the same workspace.

**Reply by dbasilio** [reply]
Yep I can put up a PR!

**Reply by dbasilio** [reply]
Posted https://github.com/TanStack/query/pull/5833, will close this discussion
